src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -1167,33 +1317,51 @@ private void addYearToStudyAdditionalInfo(Program program, BrAPIStudy study, Str
+        String envLocationName = hasAllReferenceUnitIds ?
+                pendingObsUnitByOUId.get(importRow.getObsUnitID()).getBrAPIObject().getLocationName() : importRow.getEnvLocation();
+        if (!locationByName.containsKey((importRow.getEnvLocation()))) {
+            ProgramLocation newLocation = new ProgramLocation();
+            newLocation.setName(envLocationName);
             pio = new PendingImportObject<>(ImportObjectState.NEW, newLocation, UUID.randomUUID());
-            this.locationByName.put(importRow.getEnvLocation(), pio);
+            this.locationByName.put(envLocationName, pio);
         }
     }
 
-    private PendingImportObject<BrAPITrial> fetchOrCreateTrialPIO(Program program, User user, boolean commit, ExperimentObservation importRow, Supplier<BigInteger> expNextVal) throws UnprocessableEntityException {
+    private PendingImportObject<BrAPITrial> fetchOrCreateTrialPIO(
+            Program program,
+            User user,
+            boolean commit,
+            ExperimentObservation importRow,
+            Supplier<BigInteger> expNextVal
+    ) throws UnprocessableEntityException {
         PendingImportObject<BrAPITrial> trialPio;
-        if (trialByNameNoScope.containsKey(importRow.getExpTitle())) {
-            PendingImportObject<BrAPIStudy> envPio;
-            trialPio = trialByNameNoScope.get(importRow.getExpTitle());
-            envPio = this.studyByNameNoScope.get(importRow.getEnv());
-            if  (trialPio!=null &&  ImportObjectState.EXISTING==trialPio.getState() && (StringUtils.isBlank( importRow.getObsUnitID() )) && (envPio!=null && ImportObjectState.EXISTING==envPio.getState() ) ){
-                throw new UnprocessableEntityException(PREEXISTING_EXPERIMENT_TITLE);
-            }
-        } else if (!trialByNameNoScope.isEmpty()) {
-            throw new UnprocessableEntityException(MULTIPLE_EXP_TITLES);
+
+        // use the prior trial if observation unit IDs are supplied
+        if (hasAllReferenceUnitIds) {
+            trialPio = getSingleEntryValue(trialByNameNoScope, MULTIPLE_EXP_TITLES);
+
+        // otherwise create a new trial, but there can be only one allowed
         } else {
-            UUID id = UUID.randomUUID();
-            String expSeqValue = null;
-            if (commit) {
-                expSeqValue = expNextVal.get().toString();
+            if (trialByNameNoScope.containsKey(importRow.getExpTitle())) {
+                PendingImportObject<BrAPIStudy> envPio;
+                trialPio = trialByNameNoScope.get(importRow.getExpTitle());
+                envPio = studyByNameNoScope.get(importRow.getEnv());
+                if  (trialPio!=null &&  ImportObjectState.EXISTING==trialPio.getState() &&
+                        (StringUtils.isBlank( importRow.getObsUnitID() )) && (envPio!=null && ImportObjectState.EXISTING==envPio.getState() ) ){
+                    throw new UnprocessableEntityException(PREEXISTING_EXPERIMENT_TITLE);
+                }
```

Comment:
This conditional is pretty complex, a comment explaining the intention might be helpful.